src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,142 @@
+     * 
+     * @param words Description given by user input
+     * @param type The type of task
+     * @return true if task has been added successfully, else false
+     * @throws EmptyDescriptionException when user gives an empty description
+     */ 
+    public boolean addTask(String[] words, String type) throws EmptyDescriptionException {
+        Task task;
+        if (words.length == 0) {
+            throw new EmptyDescriptionException("The description of " + type + " cannot be empty. Please try again");
+        }
+        if (type.equals("TODO")) {
+            task = new Todo(String.join(" ", words ));
+        } else {
+            String description;
+            if (type.equals("DEADLINE")) {
+                int indexForBy = Parser.getIndexOfWord(words, "/by");
+                if (indexForBy == 0) {
+                    throw new EmptyDescriptionException("The description of " + type + " cannot be empty. Please try again");
+                }
+                LocalDateTime dateTimeBy = DateTime.getDateTime(words,indexForBy);
+                if (dateTimeBy == null) {
+                    System.out.println("Please enter in this format {description} /by DD/MM/YYYY HHMM. Try again");
+                    return false;
+                }
+                description = String.join(" ",Arrays.copyOfRange(words, 0, indexForBy));
+                task = new Deadline(description,DateTime.getDateTime(words,indexForBy));
+            }
+            else {
+                int indexForFrom = Parser.getIndexOfWord(words, "/from");
+                if (indexForFrom == 0) {
+                    throw new EmptyDescriptionException("The description of " + type + " cannot be empty. Please try again");
+                }
+                int indexForTo = Parser.getIndexOfWord(words, "/to");
+
+                if (DateTime.getDateTime(words,indexForFrom) == null || DateTime.getDateTime(words,indexForTo) == null) {
+                    System.out.println("Please enter in this format {description} /from DD/MM/YYYY HHMM /to DD/MM/YYYY HHMM. Try again");
+                    return false;
+                }
+
+                LocalDateTime dateTimeFrom = DateTime.getDateTime(words,indexForFrom);
+                LocalDateTime dateTimeTo = DateTime.getDateTime(words,indexForTo);
+                description = String.join(" ",Arrays.copyOfRange(words, 0, indexForFrom));
+                task = new Event(description,dateTimeFrom,dateTimeTo);
+            }
+
+        }
+        tasks.add(task);
+        System.out.println("Got it. I've added this task:\n" + task);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+        return true;
+    }
+
+    /**
+     * Deletes a Task from the list given the index
+     * 
+     * @param num Index of the task to be deleted
+     * @return true if task has been deleted, else false
+     * @throws IndexOutOfBoundsExceptio If num > listOfTask.size()
```

Comment:
Just pointing out a very minor error. 
```suggestion
     * @throws IndexOutOfBoundsException If num > listOfTask.size()
```